src/main/java/net/minecraftforge/client/event/ScreenOpenEvent.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client.event;
+
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.eventbus.api.Cancelable;
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraftforge.fml.LogicalSide;
+
+/**
+ * Fired before any {@link Screen} will be opened, to allow changing or preventing it from being opened.
```

Comment:
"to allow changing `it` or preventing it from being opened" would make the sentence a bit more clear I think

Reply:
I feel like the current wording is clear enough, as `changing` in the sentence has no other possible subject than `it` (which would refer to the screen).

Reply:
It's more that the way I read it now, is "to allow changing it from being opened or preventing it from being opened"

Reply:
Fair point.